test/functional/wallet_listdescriptors.py

```diff
@@ -73,6 +76,17 @@ def run_test(self):
         }
         assert_equal(expected, wallet.listdescriptors())
         assert_equal(expected, wallet.listdescriptors(False))
+        assert_equal(wallet.listdescriptors({'private': False}), wallet.listdescriptors())
+        assert_equal(wallet.listdescriptors({'private': True}), wallet.listdescriptors(True))
+
+        desc_file = os.path.join(node.datadir, 'desc.json')
+        assert_equal({
+            'wallet_name': 'w2',
+            'descriptor_file': desc_file
+            }, wallet.listdescriptors({'save_to_file': desc_file}))
+
+        with open(desc_file, encoding="utf8") as json_file:
+            assert_equal(expected['descriptors'], json.load(json_file))
 
```

Comment:
maybe add a case when the file already exists?

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/25747/commits/c8840eb855bf9b5ec0b31d7b544442551dd7955f.